stan/math/rev/meta/is_vari.hpp

```diff
@@ -13,15 +13,22 @@ struct is_vari_impl : std::false_type {};
 
 template <typename T>
 struct is_vari_impl<math::vari_value<T>> : std::true_type {};
+
+template <typename T>
+struct is_vari_impl<math::vari_view<T>> : std::true_type {};
+
 }  // namespace internal
 /** \ingroup type_trait
  * Specialization for checking if value of T minus cv qualifier and pointer is a
  * vari_value.
  */
 template <typename T>
-struct is_vari<T,
-               std::enable_if_t<internal::is_vari_impl<std::decay_t<T>>::value>>
+struct is_vari<T, require_t<internal::is_vari_impl<std::decay_t<T>>>>
     : std::true_type {};
 
+template <typename T>
+struct value_type<T, require_t<is_vari<T>>> {
```

Comment:
`value_type` should be defined in value_type.hpp header!

Reply:
So for specializations over certain types we tend to define the `type_trait` that let's us identify a certain type, then write the `value_type` overload. For instance in `is_eigen.hpp` we have. 

```
template <typename T>
struct value_type<T, std::enable_if_t<is_eigen<T>::value>> {
  using type = typename std::decay_t<T>::Scalar;
};
```

So I think this should be moved over to `is_vari.hpp` in prim.

Reply:
Which I think is messy and makes the code hard to find. We should fix that. The fix might require some forward declarations, but that the specializations will not be in random files.

Reply:
I'm v cool with fixing it but let's do all of them in a different PR